libavutil/hwcontext_qsv.c

```diff
@@ -806,9 +806,19 @@ static int qsv_frames_derive_from(AVHWFramesContext *dst_ctx,
 #if CONFIG_D3D11VA
     case AV_HWDEVICE_TYPE_D3D11VA:
         {
+            dst_ctx->initial_pool_size = src_ctx->initial_pool_size;
             AVD3D11VAFramesContext *dst_hwctx = dst_ctx->hwctx;
-            mfxHDLPair *pair = (mfxHDLPair*)src_hwctx->surfaces[i].Data.MemId;
-            dst_hwctx->texture = (ID3D11Texture2D*)pair->first;
+            dst_hwctx->texture_infos = av_calloc(src_hwctx->nb_surfaces,
+                                            sizeof(*dst_hwctx->texture_infos));
+            for (i = 0; i < src_hwctx->nb_surfaces; i++) {
```

Comment:
I think we should move this loop after this line
`dst_hwctx->BindFlags = qsv_get_d3d11va_bind_flags(src_hwctx->frame_type);`

Reply:
ok I will update